src/txtempus.cc

```diff
@@ -157,7 +158,7 @@ int main(int argc, char *argv[]) {
       ttl = atoi(optarg);
       break;
     case 's':
-      service = strdup(optarg);
+      service = optarg;
```

Comment:
I think we can even avoid assigning the service string here and just directly call CreateTimeSourceFromName() here.

So in line 140, we'd have
```
   std::unique_ptr<TimeSignalSource> time_source;
  //...
  case 's':
     time_source = CreateTimeSourceFromName(optarg);
```
That way, we don't even have to worry about the service string

(Also, I's probably just leave CreateTimeSourceFromName() as is, returning a raw pointer, and assign it here as `time_source.reset(CreateTimeSourceFromName(optarg))`)